DataScience/IntroToPython/IntroToPythonPandas/blog.md

```diff
@@ -0,0 +1,215 @@
+
+## Solution
+
+```py
+salary = data.sort_values(["salary"])
+
+bottom = salary.head(6)
+top = salary.tail(6)
+```
+
+So you would get 200,000 for the 6 bottom salaries, but the top 6 salaries were NAN. That doesn’t tell us what we really want. We can filter out NULL data (like here where there was no offer made):
+```py
+salary.dropna(subset=["salary"], inplace = True)
+```
+Dropna() gets rid of null values, and arguments subset means we can remove NULL values in a certain column (here we use salary). Inplace is to choose between overwriting the same object or create a new one (true = overwrite).
+
+## Revised Solution
+
+```py
+salary = data.sort_values(["salary"])
+
+salary.dropna(subset-["salary"], inplace = True)
+bottom = salary.head(6)
+top = salary.tail(6)
+```
+
+## Altering Our Data
+
+!["Alter"](https://data-flair.training/blogs/wp-content/uploads/sites/2/2019/06/Aggregation-and-Grouping-in-Pandas.jpg)
+
+
+Say I want the average value of a certain set, or maybe I just want the highest percentage from each column. Pandas is equipped with various commands such as the ones above that can help us query the datasaet for relevant information.
+
+data.max() and data.min() will return the highest/lowest value of every column, 
+data[‘col1’].max() returns highest value in derived column. Similar implementation for .mean(), .median(), .sum(), .count(), .std() [standard deviation], and more. 
+
+For our data set:
+
+data.max() = 15 values
+
+data[‘Salary’].max() = 1 value
+
+data.mean() = 7 values, since the other 8 columns are non-numerical
+
+
+## Exercise 2
+
+Let’s get the median and sum of the top 100 salaries, as well as the max mba_p from those 100 candidates
+
+Tip: How are the salaries sorted right now?
+
+## Solution
+
+```py
+salary = data.sort_values(["salary"],ascending=False)
+
+hundred = salary.head(100)
+sum_hun = hundred["salary"].sum()
+med_hun = hundred["salary"].median()
+max_mba = hundred["mba_p"].max()
+```
+## Merging New Data
+
+What if I want to see how all the percentages add up and take the mean? Maybe that can be done with the tools already given, but now I want to add it to our original set as a new column.
+
+There are two ways to add data: **.append()** and **.concat().** Append is to add a dataframe with identical row indexes (same column values), while .concat() adds identical column indexes (same row values).
+
+Code Format:
+data1 = To be added
+data2 = Data we want to update
+data1 = data1.append(data2)
+data.concat([data1,data2])
+Add two columns with:
+New_data = data[col1] + data[col2]
+Note: This creates a Series object, not a DataFrame
+Add a series to DataFrame by creating new column:
+data[“New Col”] = series
+
+Example:
+```py
+one = pd.DataFrame({'A': [1], 'B': [3]})
+two = pd.DataFrame({'A': [2], 'B': [4]})
+one = one.append(two,ignore_index=True)
+```
+2 size(1,2) DataFrames merge into a (2,2) new object. Ignore_index works like inplace earlier.
+
+## Final Exercise Part 1
```

Comment:
Hey you did a great job with the final exercise part! However, since this is a technical blog, you can omit stuff like this and just walk us through the exercise and explain step by step how you came up with the answer.